src/distributions/binomial.rs

```diff
@@ -87,52 +94,157 @@ impl Distribution<u64> for Binomial {
             }
             result = x;
         } else {
-            // FIXME: Using the BTPE algorithm is probably faster.
-
-            // prepare some cached values
-            let float_n = self.n as f64;
-            let ln_fact_n = log_gamma(float_n + 1.0);
-            let pc = 1.0 - p;
-            let log_p = p.ln();
-            let log_pc = pc.ln();
-            let expected = self.n as f64 * p;
-            let sq = (expected * (2.0 * pc)).sqrt();
-            let mut lresult;
-
-            // we use the Cauchy distribution as the comparison distribution
-            // f(x) ~ 1/(1+x^2)
-            let cauchy = Cauchy::new(0.0, 1.0);
+            // Use the BTPE algorithm.
+
+            // Threshold for using the squeeze algorithm. This can be freely
+            // chosen based on performance. Ranlib and GSL use 20.
+            const SQUEEZE_THRESHOLD: i64 = 20;
+
+            // Step 0: Calculate constants as functions of `n` and `p`.
+            let n = self.n as f64;
+            let np = n * p;
+            let npq = np * q;
+            let f_m = np + p;
+            let m = f_m as i64;
```

Comment:
Yeah, it's similar to the GSL implementation in that regard. (However, I used the paper as a reference for my implementation to avoid issues with the GPL.) Using integers should be faster and more precise, but I'll look into simplifying the code.

> Out-of-range cast from float to int is undefined.

I'll add a check that this panics instead.